fileconnector/fileconnector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FileCopy.java

```diff
@@ -0,0 +1,197 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.connector;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import javax.xml.stream.XMLStreamException;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.vfs2.*;
+import org.apache.commons.vfs2.impl.StandardFileSystemManager;
+import org.apache.synapse.MessageContext;
+import org.codehaus.jettison.json.JSONException;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.Connector;
+import org.wso2.carbon.connector.core.util.ConnectorUtils;
+import org.wso2.carbon.connector.util.FileConnectorUtils;
+import org.wso2.carbon.connector.util.FileConstants;
+import org.wso2.carbon.connector.util.FilePattenMatcher;
+import org.wso2.carbon.connector.util.ResultPayloadCreate;
+
+public class FileCopy extends AbstractConnector implements Connector {
+    private static final Log log = LogFactory.getLog(FileCopy.class);
+
+    public void connect(MessageContext messageContext) {
+        String source = (String) ConnectorUtils.lookupTemplateParamater(messageContext,
+                FileConstants.FILE_LOCATION);
+        String destination = (String) ConnectorUtils.lookupTemplateParamater(messageContext,
+                FileConstants.NEW_FILE_LOCATION);
+        String filePattern = (String) ConnectorUtils.lookupTemplateParamater(messageContext,
+                FileConstants.FILE_PATTERN);
+        FileSystemOptions opts = FileConnectorUtils.init(messageContext);
+        boolean resultStatus = copyFile(source, destination, filePattern, messageContext, opts);
+        ResultPayloadCreate resultPayload = new ResultPayloadCreate();
+        generateResults(messageContext, resultStatus, resultPayload);
+    }
+
+    /**
+     * Generate the results
+     *
+     * @param messageContext The message context that is processed by a handler in the handle method
+     * @param resultStatus   Result of the status (true/false)
+     * @param resultPayload  result payload create
+     */
+    private void generateResults(MessageContext messageContext, boolean resultStatus,
+                                 ResultPayloadCreate resultPayload) {
+        String response = FileConstants.START_TAG + resultStatus + FileConstants.END_TAG;
+        OMElement element;
+        try {
+            element = resultPayload.performSearchMessages(response);
+            resultPayload.preparePayload(messageContext, element);
+        } catch (XMLStreamException e) {
+            log.error(e.getMessage());
```

Comment:
Jus logging the stack-trace would not make much sense. Could you add a meaningful error log ?

Reply:
https://github.com/wso2-extensions/esb-connector-file/pull/28